augur/io/metadata.py

```diff
@@ -111,7 +111,10 @@ def read_metadata(metadata_file, delimiters=DEFAULT_DELIMITERS, id_columns=DEFAU
     # of having its type inferred. This latter argument allows users to provide
     # numerical ids that don't get converted to numbers by pandas.
     kwargs["index_col"] = index_col
-    kwargs["dtype"] = {index_col: "string"}
+    kwargs["dtype"] = {
+        index_col: "string",
+        METADATA_DATE_COLUMN: "string"
+    }
```

Comment:
_non-blocking_

I kinda wonder if we couldn't just treat everything as a string? Do we need other types inferred? (I think I've wondered this before in another PR too…?)

Reply:
Hmm, would this be an issue for `augur export` if a user sets a metadata column as a `continuous` coloring? 
We also try to guess coloring type within `augur export` if the trait value type is not provided: 
https://github.com/nextstrain/augur/blob/b61e3e7e969ff1b82fce5f2e2f388a10e6f3c305/augur/export_v2.py#L239-L245

Reply:
Nod, yeah, we'd have to fiddle with that. Instead of testing if the values are all already floats, it would instead test if it could convert all values to `float`.

Separately, there's probably some metadata types that get inferred from CSV as non-string and then re-serialized to JSON with those non-string types.

Reply:
(In any case, this isn't a "this PR" change!)

Reply:
#1252